app/build.gradle

```diff
@@ -33,11 +56,47 @@ android {
 
 dependencies {
 
+    // Import the BoM for the Firebase platform
+    implementation platform('com.google.firebase:firebase-bom:29.0.1')
+
+    // Declare the dependency for the Firebase Authentication library
+    // When using the BoM, you don't specify versions in Firebase library dependencies
+    implementation 'com.google.firebase:firebase-auth'
+
+    // Also declare the dependency for the Google Play services library and specify its version
+    implementation 'com.google.android.gms:play-services-auth:20.0.0'
+
+    implementation 'com.google.firebase:firebase-analytics:20.0.0'
+    implementation 'com.google.firebase:firebase-auth-ktx:21.0.1'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    //Ktlint
+    ktlint 'com.pinterest:ktlint:0.43.2'
+
+    //Firebase
+    implementation "com.google.firebase:firebase-auth:21.0.1"
+    //Firebase: Analytics, DB, Crashlytics, Notifications
+    implementation 'com.google.firebase:firebase-analytics-ktx:20.0.0'
+    implementation 'com.google.firebase:firebase-database-ktx:20.0.3'
+    implementation 'com.google.firebase:firebase-crashlytics-ktx:18.2.5'
+    implementation 'com.google.firebase:firebase-messaging-ktx:23.0.0'
+
+    //for kotlin use kapt!!!
+    kapt "androidx.room:room-compiler:2.3.0"
```

Comment:
I don't see `Room` dependencies - why did you add this without main Room dependency?